pkg/controller/resource_sync_task.go

```diff
@@ -379,17 +420,6 @@ func (st *resourceSyncTask) createResource(resClient dynamic.ResourceInterface,
 
 // Mutates spec and actual.
 func (st *resourceSyncTask) updateResource(resClient dynamic.ResourceInterface, spec *unstructured.Unstructured, actual runtime.Object) (actualRet *unstructured.Unstructured, retriableError bool, e error) {
-	actualMeta := actual.(meta_v1.Object)
-	// Check that the object is not marked for deletion
-	if actualMeta.GetDeletionTimestamp() != nil {
-		return nil, false, errors.New("object is marked for deletion")
-	}
-
-	// Check that this bundle owns the object
-	if !meta_v1.IsControlledBy(actualMeta, st.bundle) {
-		return nil, false, errors.New("object is not owned by the Bundle")
-	}
-
```

Comment:
Moved to `getActualObject()` to do these checks earlier. Because of that the integration tests had to be updated - new (better!) behaviour.